pkg/kubelet/server/remotecommand/exec.go

```diff
@@ -51,8 +65,16 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
 
 	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
 	if err != nil {
-		msg := fmt.Sprintf("error executing command in container: %v", err)
-		runtime.HandleError(errors.New(msg))
-		fmt.Fprint(ctx.errorStream, msg)
+		m := ExitError{}
+		if exitErr, ok := err.(utilexec.ExitError); ok {
+			m.Message = exitErr.Error()
+			m.ExitCode = exitErr.ExitStatus()
+			m.Exited = true
+		} else {
+			m.Message = fmt.Sprintf("error executing command in container: %v", err)
+		}
+		runtime.HandleError(errors.New(m.Message))
+		bs, _ := json.Marshal(&m)
```

Comment:
not really much we can do here in case of an error. Added a return on err for uniformity.